nodejs/eks/nodegroup.ts

```diff
@@ -1544,6 +1544,43 @@ export type ManagedNodeGroupOptions = Omit<
      *   - maxSize: 2
      */
     scalingConfig?: pulumi.Input<awsInputs.eks.NodeGroupScalingConfig>;
+
+    /**
+     * The AMI type for the instance.
+     *
+     * If you are passing an amiId that is `arm64` type, then we need to ensure
+     * that this value is set as `amazon-linux-2-arm64`.
+     *
+     * Note: `amiType` and `gpu` are mutually exclusive.
+     */
+    amiType?: pulumi.Input<string>;
```

Comment:
These looks like something copied over from NodeGroupBaseOptions, I wonder if we could factor this out in TypeScript somehow.

Reply:
I had a look, this inherits directly from the AWS config so I figured we shouldn't mess with it too much.